src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -574,31 +574,32 @@ GenericParameterValue genericParam
 				// instanceValue here is like argumentValues[0] in linker
 				foreach (var value in instanceValue) {
 					// Nullables without a type argument are considered SystemTypeValues
-					if (value is SystemTypeValue typeValue
-							&& typeValue.RepresentedType.IsTypeOf ("System", "Nullable`1")) {
-						foreach (var argumentValue in argumentValues[0]) {
-							if ((argumentValue as ArrayValue)?.TryGetValueByIndex (0, out var underlyingMultiValue) == true) {
-								foreach (var underlyingValue in underlyingMultiValue) {
-									switch (underlyingValue) {
-									case SystemTypeValue systemTypeValue:
-										AddReturnValue (new NullableSystemTypeValue (typeValue.RepresentedType, systemTypeValue.RepresentedType));
-										break;
-									case ValueWithDynamicallyAccessedMembers damValue:
-										AddReturnValue (new NullableValueWithDynamicallyAccessedMembers (typeValue.RepresentedType, damValue));
-										break;
-									// Nullable values and array values cannot be used as generic arguments to nullables, so we don't need to worry about anything else here
-									default:
-										AddReturnValue (MultiValueLattice.Top);
-										break;
-									}
-								}
+					if (!(value is SystemTypeValue typeValue && typeValue.RepresentedType.IsTypeOf ("System", "Nullable`1"))) {
+						// We still don't want to lose track of the type if it is not Nullable<T>
+						AddReturnValue (value);
+						continue;
+					}
+					// We have a nullable
+					foreach (var argumentValue in argumentValues[0]) {
+						if ((argumentValue as ArrayValue)?.TryGetValueByIndex (0, out var underlyingMultiValue) != true) {
+							continue;
+						}
+						foreach (var underlyingValue in underlyingMultiValue) {
+							switch (underlyingValue) {
+							case SystemTypeValue systemTypeValue:
+								AddReturnValue (new NullableSystemTypeValue (typeValue.RepresentedType, systemTypeValue.RepresentedType));
+								break;
+							case ValueWithDynamicallyAccessedMembers damValue:
+								AddReturnValue (new NullableValueWithDynamicallyAccessedMembers (typeValue.RepresentedType, damValue));
+								break;
+							// Nullable values and array values cannot be used as generic arguments to nullables, so we don't need to worry about anything else here
+							default:
+								AddReturnValue (MultiValueLattice.Top);
+								break;
```

Comment:
I think we need to be a bit more careful. We should special case array and nullable and add Top to the return, but everything else should probably add the "Return value of MakeGenericType" with no annotations. Basically we can't skip over unknown values:
```C#
void Test(object[] maybetypes)
{
    Type[] types = new Type[] { maybetypes[0] as Type };  // Roundabout way to get UnknownValue - it is getting tricky to do that reliably
    Type nullable = typeof (Nullable<>).MakeGenericType(types);
    nullable.GetProperties();  // Must WARN
}
```